### ISSUE TITLE
uBO prevents loading of web pages on new windows if an address is entered too quickly

### ISSUE THREAD

**Issue by DrCta** [uBlockOrigin/uBlock-issues#1944]
### Prerequisites

- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] using a new, unmodified browser profile

### Description

In Chrome, when a normal or incognito window is opened _whilst_ there are no other windows _of the same type_ already existing, uBO prevents the normal loading of webpages that are requested before the window has time to fully 'load' itself. Very few domains from Google appear to bypass this, such as [Google Maps](https://www.google.com/maps). All other sites I've tried suffer from this.

**Disabling uBO** on the destination site or on the 'New Tab' page makes no difference. The issue is only gone once uBO is disabled through the extensions manager on Chrome. Incognito windows don't suffer from this if uBO is not allowed to run in incognito. Also, disabling it on the New Tab page will make it so the icon goes from red to gray a moment after the loading is interrupted.

### A specific URL where the issue occurs

https://login.live.com/

### Steps to Reproduce

1. Close all incognito windows if there are any.
2. Open an incognito window (through any means) and **immediately** do any one of the following:
    **a.** Click on a site in your bookmarks
    **b.** Type a search term in the address bar and press Enter
    **c.** Type the initials of a site your browser autofills in the address bar and press Enter
    **d.** Paste a site into the address bar from your clipboard and press Enter

### Expected behavior

As per the behaviour when uBO is disabled, the newly opened window would stagger (not do anything) very briefly until it finishes launching itself, then hiding the New Tab's elements and starting to load the requested webpage or search results quite normally.

### Actual behavior

With uBO enabled, this 'stagger' is still seen, but at the end of it, the background on the New Tab page blinks, the address bar is cleared and no further events occur. The requested content is not loaded at all.

### Configuration

<details>

```yaml
uBlock Origin: 1.40.8
Chromium: 89
filterset (summary): 
  network: 79105
  cosmetic: 60584
  scriptlet: 16517
  html: 0
listset (total-discarded, last updated): 
  added: 
    fanboy-cookiemonster: 21516-15, 8d.15h.30m
  default: 
    easylist: 63173-9, 3d.19h.57m
    easyprivacy: 26637-17, 3d.19h.56m
    plowe-0: 3682-816, 2d.16h.14m
    ublock-abuse: 76-0, 3d.19h.54m
    ublock-badware: 3724-67, 3d.19h.53m
    ublock-filters: 30203-156, 3d.19h.51m
    ublock-privacy: 190-3, 3d.19h.50m
    ublock-unbreak: 1683-40, 3d.19h.49m
    urlhaus-1: 6565-0, 5h.22m
filterset (user): [array of 29 redacted]
trustedset: 
  added: [array of 51 redacted]
  removed: 
    chrome-extension-scheme
    chrome-scheme
    edge-scheme
switchRuleset: 
  added: [array of 1 redacted]
modifiedUserSettings: [none]
modifiedHiddenSettings: [none]
supportStats: 
  launchToReadiness: 1133 ms (selfie)
```
</details>

**Comment by uBlock-user** [comment]
Disable _Suspend network activity until all filter lists are loaded_ in Filterlists pane and try again.

**Comment by gwarser** [comment]
Are you launching from spinning disk hard drive? (so it will take more time for browser and uBO to launch)

Is your network latency very high? (so any connection will not actually happen before uBO will reload active tab)

uBO should not block anything on startup by default, but when it's ready it will reload active tab. I think this may cause issue like this when it happens with specific timing. (eg. type `example.com` in address bar and then press f5 - typed content will disappear)

**Comment by DrCta** [comment]
Thank you for the reply.
I am running Windows 7 on an SSD and on very decent hardware.
The latency to the sites I've tested appear to sit between 250 and 150ms.

I've looked into that setting as it wasn't easily accessible through the filter lists panel, and did a few more experiments.

This setting was "_unset_". Changing it to "_yes_" yielded three outcomes seemingly at random:

- Website loads normally after that short delay
- The New Tab page goes away immediately and shows a "website is blocked by extension" error page, before refreshing and loading the requested content normally. When doing Google searches, the results page gets loaded partially, then refreshed, and loaded normally.
- The address bar is **very** quickly cleared (much faster than before) and no content is loaded.

When set to "_no_", the requests **were fulfilled normally**. Sites and search queries loaded per usual. I did not notice any anomalies in the sites themselves, no ads displayed or such.

And indeed, pressing F5 on the New Tab page mimics the exact blinking I've witnessed and also clears the address bar before it has a chance to load properly.

**Comment by gorhill** [comment]
The issue should be gone in current dev build, or at least quite less likely. In any case, even if this is reproducible in current dev build, the only solution is for the user to disable the setting _"Suspend network activity until all filter lists are loaded"_. There no way for uBO to know that the user is currently editing the URL in the address bar.

**Comment by Dorennor** [comment]
I have same issue.

**Comment by kenktk** [comment]
I also have the same issue. 
Hardware:
Chromebook 11 3810
Coreboot

Software:
Lubuntu 21.10
Ungooged Chromium 95

I have been able to replicate the issue several times on two of the same machines. Removing the extension resolves the issue. The UI of the extension does not load when clicked (this may be related to another issue, perhaps the slow rendering of the UI per my subpar hardware).

**Comment by gorhill** [comment]
As there is no feedback regarding comment above, closing as assumed fixed with <https://github.com/gorhill/uBlock/commit/925c8d5d0c37dbc1f82e57a92e74350de2c5eab1>.
